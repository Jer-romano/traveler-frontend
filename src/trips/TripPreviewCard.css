.TripPreviewCard {
  max-height: 500px;
  max-width: 700px;
  margin-bottom: 1rem;
  color: inherit;
  border-radius: 0;
  border: 1px solid grey;
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
}

.TripPreviewCard img {
  max-height: 300px;
}

.TripPreviewCard:hover { /* <Card /> that is also <Link /> */
  text-decoration: none;
  box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16), 0 3px 6px rgba(0, 0, 0, 0.23);
}

.resp-img {
  max-width: 800px;
}