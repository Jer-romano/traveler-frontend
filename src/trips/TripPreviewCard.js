import React from "react";
import { Link } from "react-router-dom";
import Tag from "./Tag";
import "./TripPreviewCard.css";

/** Show limited information about a trip
 *
 * Is rendered by TripList to show a "card" for each trip.
 *
 * TripList -> TripPreviewCard
 */

function TripPreviewCard({id, title, tripImage, username }) {
  console.debug("TripPreviewCard", id);

  return (
      <Link className="TripPreviewCard card" to={`/trips/${id}`}>
        <img src={tripImage.fileUrl}
                             alt={"A picture of the trip"}
                             className="card-img-top" />
        <div className="card-body">
           <h2 className="card-title">
           {title}
          </h2>
   
          <span className="mt-4">
            <Tag tag={tripImage.tag1}/>
            <Tag tag={tripImage.tag2}/>
            <Tag tag={tripImage.tag3}/>
          </span>
          <h4 className="mt-4">
            {/* {profileImage && <img src={profileImage}
                             alt={"A picture of the user"}
                             className="float-left ml-5" />} */}
            {username && `Posted by ${username}`}
          </h4>
        </div>
      </Link>
  );
}

export default TripPreviewCard;
