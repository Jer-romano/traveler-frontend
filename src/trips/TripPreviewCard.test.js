import React from "react";
import { render } from "@testing-library/react";
import TripPreviewCard from "./TripPreviewCard";
import { MemoryRouter } from "react-router";

// it("renders without crashing", function () {
//     render(<TripPreviewCard/>);
// });

const img = {
    fileUrl: "../images/brazil_photo2.jpeg",
    tag1: "Cool",
    tag2: "Awesome",
    tag3: "Sky",
    tag4: "Cloud",
    tag5: "Building"
};


it("matches snapshot", function() {
    const { asFragment } = render(
    <MemoryRouter>
        <TripPreviewCard
        id={1}
        title={"The best trip"}
        tripImage={img}
        username={"testuser"}
        />
    </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});