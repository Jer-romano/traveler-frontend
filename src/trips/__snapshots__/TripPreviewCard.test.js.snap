// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <a
    class="TripPreviewCard card"
    href="/trips/1"
  >
    <img
      alt="The trip"
      class="card-img-top"
      src="../images/brazil_photo2.jpeg"
    />
    <div
      class="card-body"
    >
      <h2
        class="card-title"
      >
        The best trip
      </h2>
      <span
        class="mt-4"
      >
        <span
          class="tag"
        >
          Cool
        </span>
        <span
          class="tag"
        >
          Awesome
        </span>
        <span
          class="tag"
        >
          Sky
        </span>
        <span
          class="tag"
        >
          Cloud
        </span>
        <span
          class="tag"
        >
          Building
        </span>
      </span>
      <h4
        class="mt-4"
      >
        Posted by testuser
      </h4>
    </div>
  </a>
</DocumentFragment>
`;
